src/controllers/User.controller.ts

```diff
@@ -63,6 +68,8 @@ export const createUser = async (req: Request, res: Response) => {
   } catch (error) {
     if (error instanceof Error) {
```

Comment:
```suggestion
catch (error) {
    if (error instanceof Error) {
      console.error(`Error creating user: ${error.message}`, error);
      res.status(500).json({
        status: 500,
        success: false,
        message: `Error creating user: ${error.message}`,
      });
    } else {
      console.error("Unexpected error occurred", error);
      res.status(500).json({
        status: 500,
        success: false,
        message: "Unexpected error occurred",
      });
    }
  }
```
@SergeInGithub  I'd suggest you change the whole catch block to the suggested code above, that way it would be easier to read and would generate a clean error message with fewer lines of code